src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -119,32 +130,53 @@ class PalantirDockerPlugin implements Plugin<Project> {
                 logging.captureStandardError LogLevel.ERROR
             }
 
-            if (!ext.tags.isEmpty()) {
+            Map<String, Object> tags = ext.tags.collectEntries { taskName, tagName ->
+                [generateTagTaskName(taskName), [
+                        rawTagName          : tagName,
+                        resolveTagNameAction: { -> tagName }
+                ]]
+            }
+
+            if (!ext.unresolvedTags.isEmpty()) {
+                ext.unresolvedTags.each { unresolvedTagName ->
+                    String taskName = generateTagTaskName(unresolvedTagName)
+
+                    if (tags.containsKey(taskName)) {
+                        throw new GradleException("Task name '${taskName}' of docker tag '${unresolvedTagName}' is existed.")
+                    }
+
+                    tags[taskName] = [
+                            rawTagName          : unresolvedTagName,
+                            resolveTagNameAction: { -> computeName(ext.name, unresolvedTagName) }
```

Comment:
Why make this a closure? We're almost certainly going to evaluate it so I'd prefer to just eagerly evaluate it to make this simpler.

Reply:
Due to this test 'does not throw if name is configured after evaluation phase'.
```Groovy
docker {
    tags 'latest', 'another', 'withTaskName@2.0', 'newImageName@${id}-new:latest'
    tag 'withTaskNameByTag', '${id}:new-latest'
}
afterEvaluate {
    docker.name = '${id}'
}
```

You can set the docker name in `afterEvaluate`, if we resolve the tag immediately, we could not get the name of the docker, and the name of docker is required for resolve tag, so it will throw an exception.
So, I create a closure for resolving it.